### PR TITLE
Closes #374 - Audit logging for configuration server.

### LINKED ISSUE #374: Provide audit log for the configuration server
Many users require to have some kind of audit log for each components which can be configured and modified by end-users.

Due to this reason, the configuration-server should log each modification of any kind of data (agent mappings, configuration changes, user added, ...) in the normal log stream. The log entry should contain the used username and the type of data which has been changed.
If availalbe, it would also be helpful to log what was changed (e.g. certain user has been created, ...).

### PR DESCRIPTION
**Pull request by abhishekb03**
- Audit logging for User
- Audit logging for Agent-Mapping

---
This change is [Reviewable](https://reviewable.io/reviews/inspectit/inspectit-ocelot/434)

### CODE REVIEW

**Comment by abhishekb03** [comment]
I've added audit-logging for users and agent-mappings. Let me know what else has to be audit-logged.
Let me know if this works for you.

**Comment by JonasKunz** [comment]
Something else I forgot to mention:
Could you also add a debug level message when agents fetch the configuration to the `AgentController`?

**Comment by abhishekb03** [comment]
@JonasKunz Thank you for your feedback and suggestions. I'll modify the PR based on above issues shortly (if not fixed already).
@JonasKunz @mariusoe I was travelling in the last week and could not connect.

File: components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/security/audit/AuditEventListener.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import javax.persistence.PostPersist;
+import javax.persistence.PostRemove;
+import javax.persistence.PostUpdate;
+
+/**
+ * Listener for events to be audit logged.
+ */
+@Slf4j
+public class AuditEventListener {
```
Comment (abhishekb03) [inline components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/security/audit/AuditEventListener.java]: I have still kept EventListener as it offers more granular events for JPA based entities like User. For example, it makes distinction among CREATE & UPDATE as against SAVE.
Please let me know if you want this removed.

File: components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/security/audit/EntityAuditLogger.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@Component
+@Slf4j
+public class EntityAuditLogger {
+
+    private static final String DEFAULT_USER = "anonymousUser";
```
Comment (abhishekb03) [inline components/inspectit-ocelot-configurationserver/src/main/java/rocks/inspectit/ocelot/security/audit/EntityAuditLogger.java]: Named it to `anonymousUser` and not `anonymous` as  `anonymousUser` is being used at other places too.

**Comment by abhishekb03** [comment]
@JonasKunz Changed based on your suggestions. 
I was looking to use AOP or annotation driven logging. I'll create a PR referring to this if that is feasible.
I hope this will work till then.

**Comment by JonasKunz** [comment]
Great work! Thank you :)

**Pull request merged**